packages/ra-ui-materialui/src/field/ReferenceArrayField.js

```diff
@@ -87,38 +47,70 @@ export const ReferenceArrayField = ({ children, ...props }) => {
     return (
         <ReferenceArrayFieldView
             {...props}
-            {...useReferenceArray(props)}
+            {...useReferenceArrayFieldController(props)}
             children={children}
         />
     );
 };
 
 ReferenceArrayField.propTypes = {
+    ...fieldPropTypes,
     addLabel: PropTypes.bool,
-    basePath: PropTypes.string.isRequired,
+    basePath: PropTypes.string,
     classes: PropTypes.object,
     className: PropTypes.string,
     children: PropTypes.element.isRequired,
     label: PropTypes.string,
-    record: PropTypes.object.isRequired,
+    record: PropTypes.object,
     reference: PropTypes.string.isRequired,
-    resource: PropTypes.string.isRequired,
+    resource: PropTypes.string,
     sortBy: PropTypes.string,
     source: PropTypes.string.isRequired,
 };
 
-const EnhancedReferenceArrayField = withStyles(styles)(ReferenceArrayField);
-
-EnhancedReferenceArrayField.defaultProps = {
+ReferenceArrayField.defaultProps = {
     addLabel: true,
 };
 
-EnhancedReferenceArrayField.propTypes = {
-    ...fieldPropTypes,
-    reference: PropTypes.string,
-    children: PropTypes.element.isRequired,
+const useStyles = makeStyles({
+    progress: { marginTop: '1em' },
+});
+
+export const ReferenceArrayFieldView = ({
+    children,
+    className,
+    classes: classesOverride,
+    data,
+    ids,
+    loaded,
+    reference,
+    referenceBasePath,
+}) => {
+    const classes = useStyles({ classes: classesOverride });
+    if (loaded === false) {
+        return <LinearProgress className={classes.progress} />;
+    }
+
+    return React.cloneElement(Children.only(children), {
```

Comment:
Not important but I wonder why you don't just import `cloneElement` instead of using `React.cloneElement`